File: src/Server/UI/Variant.cs
namespace Belin.Base.UI;

using System.Text.Json.Serialization;

/// <summary>
/// Defines tone variants.
/// </summary>
[JsonConverter(typeof(JsonStringEnumConverter))]
public enum Variant {

	/// <summary>
	/// A danger.
	/// </summary>
	Danger = Context.Danger,

	/// <summary>
	/// A warning.
	/// </summary>
	Warning = Context.Warning,

	/// <summary>
	/// An information.
	/// </summary>
	Info = Context.Info,

	/// <summary>
	/// A success.
	/// </summary>
	Success = Context.Success,

	/// <summary>
	/// A primary tone.
	/// </summary>
	Primary,

	/// <summary>
	/// A secondary tone.
	/// </summary>
	Secondary,

	/// <summary>
	/// A light tone.
	/// </summary>
	Light,

	/// <summary>
	/// A dark tone.
	/// </summary>
	Dark
}

/// <summary>
/// Provides extension methods for tone variants.
/// </summary>
public static class VariantExtensions {

	/// <summary>
	/// Returns the CSS representation of the specified variant.
	/// </summary>
	/// <param name="variant">The variant.</param>
	/// <returns>The CSS representation of the specified variant.</returns>
	public static string ToCss(this Variant variant) => variant.ToString().ToLowerInvariant();
}

File: src/Client/UI/Variant.ts
import {Context} from "./Context.js";

/**
 * Defines tone variants.
 */
export const Variant = Object.freeze({
	...Context,

	/**
	 * A primary tone.
	 */
	Primary: "Primary",

	/**
	 * A secondary tone.
	 */
	Secondary: "Secondary",

	/**
	 * A light tone.
	 */
	Light: "Light",

	/**
	 * A dark tone.
	 */
	Dark: "Dark"
});

/**
 * Defines tone variants.
 */
export type Variant = typeof Variant[keyof typeof Variant];

/**
 * Returns the CSS representation of the specified variant.
 * @param variant The variant.
 * @returns The CSS representation of the specified variant.
 */
export function toCss(variant: Variant): string {
	return variant.toLowerCase();
}
